File: examples/python/core/invoke/process-document.py
@durable_execution
def process_document(event: dict, context: DurableContext) -> dict:
    """Process a document based on its type."""
    document_type = event["document_type"]
    document_data = event["data"]
    
    if document_type == "pdf":
        result = context.invoke(
            function_name="process-pdf",
            payload=document_data,
            name="process_pdf",
        )
    elif document_type == "image":
        result = context.invoke(
            function_name="process-image",
            payload=document_data,
            name="process_image",
        )
    else:
        result = context.invoke(
            function_name="process-generic",
            payload=document_data,
            name="process_generic",
        )
    
    return result

File: examples/java/core/invoke/process-document.java
public class ProcessDocumentHandler extends DurableHandler<Map<String, Object>, Map<String, Object>> {

    @Override
    public Map<String, Object> handleRequest(Map<String, Object> event, DurableContext ctx) {
        String documentType = (String) event.get("document_type");
        var documentData = event.get("data");

        Map result;
        if ("pdf".equals(documentType)) {
            result = ctx.invoke("process_pdf", "process-pdf", documentData, Map.class);
        } else if ("image".equals(documentType)) {
            result = ctx.invoke("process_image", "process-image", documentData, Map.class);
        } else {
            result = ctx.invoke("process_generic", "process-generic", documentData, Map.class);
        }

        return result;
    }
}
